pkg/common/ipsec/utils.go

```diff
@@ -15,18 +17,24 @@ const (
 	markStateOut = 0xe00
 )
 
-func CountUniqueIPsecKeys(states []netlink.XfrmState) int {
+func CountUniqueIPsecKeys(states []netlink.XfrmState) (int, error) {
 	keys := make(map[string]bool)
 	for _, s := range states {
 		if s.Aead != nil {
 			keys[string(s.Aead.Key)] = true
+			continue
 		}
-		if s.Crypt != nil {
-			keys[string(s.Crypt.Key)] = true
+		if s.Auth == nil && s.Crypt == nil {
+			continue
 		}
+		if (s.Auth != nil && s.Crypt == nil) || (s.Auth == nil && s.Crypt != nil) {
+			return 0, fmt.Errorf("invalid XfrmState found [Auth != nil -> %t] [Crypt != nil -> %t]", s.Auth != nil, s.Crypt != nil)
```

Comment:
This is a bit cryptic for an end-user. I think it would be good enough to say that "an unsupported key with only Auth or Crypt set has been found".

Reply:
Done, thanks!